src/watchmaker/utils/imds/detect/providers/aws_provider.py

```diff
@@ -71,23 +64,30 @@ def __is_valid_server(self):
                 return True
         return False
 
-    def __get_data_from_server(self):
-        """Retrieve AWS metadata."""
-        response = utils.urlopen_retry(self.metadata_url, self.DEFAULT_TIMEOUT)
-        if response.status == 200:
-            return response.read()
+    def __get_metadata_request_headers(self):
+        if AWSProvider.imds_token:
+            self.logger.debug("Making IMDSv2 Call")
+            return {"X-aws-ec2-metadata-token": AWSProvider.imds_token}
+
+        self.logger.debug("Making IMDSv1 Call")
         return None
 
-    def __get_file_contents(self, file):
-        """Get file contents if exists."""
-        if not exists(file):
-            return None
-        # Convert a local vendor file to a URI
-        file_path = utils.uri_from_filepath(file)
+    def __request_token(self):
         try:
-            check_vendor_file = utils.urlopen_retry(
-                file_path, self.DEFAULT_TIMEOUT
+            self.logger.debug("Create request for token")
+            AWSProvider.imds_token = self.__call_urlopen_retry(
+                self.metadata_imds_v2_token_url,
+                self.DEFAULT_TIMEOUT,
+                headers={"X-aws-ec2-metadata-token-ttl-seconds": "21600"},
```

Comment:
Might as well tighten this up a little, if watchmaker is running for 6 hours, it's broke anyway.

```suggestion
                headers={"X-aws-ec2-metadata-token-ttl-seconds": "7200"},
```